qiskit/passmanager/propertyset.py

```diff
@@ -10,15 +10,54 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""A property set is maintained by the pass runner.
+"""A property set dictionary that shared among optimization passes."""
 
 
-This is sort of shared memory space among passes.
-"""
+from dataclasses import dataclass, field
+from enum import IntEnum
+
+from .exceptions import PassManagerError
 
 
 class PropertySet(dict):
-    """A default dictionary-like object"""
+    """A default dictionary-like object."""
 
     def __missing__(self, key):
         return None
+
+
+class FencedPropertySet(PropertySet):
+    """A readonly property set that cannot be written via __setitem__."""
+
+    def __setitem__(self, key, value):
+        raise PassManagerError("The fenced PropertySet has the property __setitem__ protected.")
+
+
+class RunState(IntEnum):
+    """Allowed values for the result of a pass execution."""
+
+    FAIL = 0
+    SUCCESS = 1
+    SKIP = 2
```

Comment:
What sort of interface / usage were you expecting that necessitates this being a subclass of `int`?  It doesn't feel like there's any hard requirement for it to subclass `int`, and doing so allows various places to use `int` when they probably should be compelled by the type system to use `RunState` (e.g. defaults should be `RunState.SUCCESS` not `1`).

Reply:
There is no reason to choose integer. We just need some mechanism to check if pass finishes without error. This is currently only used by `TransformationPass` to add calibration to successful DAG.

Reply:
In that case, maybe it ought to be a raw `enum.Enum` to enforce that people _must_ refer to the status codes by name only?

Reply:
Fair enough. Done in [a18dd91](https://github.com/Qiskit/qiskit/pull/10127/commits/a18dd912ba5f14c7a2afc3cc81524253f38669a5)